src/Compilers/Core/Portable/SourceGeneration/SyntaxStore.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.Collections;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal sealed class SyntaxStore
+    {
+        private readonly StateTableStore _tables;
+        private readonly Compilation? _compilation;
+        internal static SyntaxStore Empty = new SyntaxStore(StateTableStore.Empty, compilation: null);
+
+        private SyntaxStore(StateTableStore tables, Compilation? compilation)
+        {
+            _tables = tables;
+            _compilation = compilation;
+        }
+
+        public Builder ToBuilder(Compilation compilation, ImmutableArray<SyntaxInputNode> syntaxInputNodes, bool enableTracking, CancellationToken cancellationToken = default) => new Builder(compilation, syntaxInputNodes, enableTracking, this, cancellationToken);
```

Comment:
>cancellationToken = default [](http://example.com/codeflow?start=147&length=27)

It looks like there is only one caller of `ToBuilder()`. Given that, consider making `cancellationToken` not optional.

Reply:
It's also used in a test, but I'll make it required and just pass it in explicitly there.